### DISCUSSION TITLE
What's the difference between commit() and commitDurable()

### DISCUSSION THREAD

**Discussion by ghost** [questdb/questdb#1137] [Ideas]
Hi,

This is not really an issue per se, but I was wondering what the implications of using `commit()` vs `commitDurable()` are. I've noticed that the number of writes per second drops to a very low figure when using the durable mode. Is this because the database doesn't really flush to disk synchronously when using simply `commit()`? In case of power failure or similar issues, would that cause any data to be lost?

**Comment by bluestreak01** [comment]
Hi, you are correct, `commit()` does not physically flush data to underlying storage, which may result in data loss if operating system fails to shutdown gracefully. In case of process or even operating system crash `commit()` is sufficient to prevent data loss.

`commitDurable()` is for exceptional cases where you have critical data on unreliable hardware.

**Comment by ghost** [comment]
That's what I wanted to know. Thanks!
